lib/ansible/utils/display.py

```diff
@@ -520,7 +591,139 @@ def do_var_prompt(self, varname, private=True, prompt=None, encrypt=None, confir
 
     def _set_column_width(self):
         if os.isatty(1):
-            tty_size = unpack('HHHH', fcntl.ioctl(1, TIOCGWINSZ, pack('HHHH', 0, 0, 0, 0)))[1]
+            tty_size = unpack('HHHH', fcntl.ioctl(1, termios.TIOCGWINSZ, pack('HHHH', 0, 0, 0, 0)))[1]
         else:
             tty_size = 0
         self.columns = max(79, tty_size - 1)
+
+    def prompt_until(self, msg, private=False, seconds=None, interrupt_input=None, complete_input=None):
+        if self._final_q:
+            from ansible.executor.process.worker import current_worker
+            self._final_q.send_prompt(
+                worker_id=current_worker.worker_id, prompt=msg, private=private, seconds=seconds,
+                interrupt_input=interrupt_input, complete_input=complete_input
+            )
+            return current_worker.worker_queue.get()
+
+        if HAS_CURSES and not self.setup_curses:
+            setupterm()
+            self.setup_curses = True
+
+        if (
```

Comment:
Don't we need to check `not HAS_CURSES` here as well?

Reply:
Why? If curses couldn't be imported or setupterm fails, we'll use the hardcoded values for MOVE_TO_BOL and CLEAR_TO_EOL.

Reply:
:facepalm: Yep, you're right- I forgot the original code was only actually using curses to query the terminfo and always doing the terminal state management manually (bleh). So it's consistent with the original code as you have it.